CRM/Admin/Form/MailSettings.php

```diff
@@ -109,11 +127,15 @@ public function getDefaultEntity() {
   public function setDefaultValues() {
     $defaults = parent::setDefaultValues();
 
-    // Set activity status to "Completed" by default.
-    if ($this->_action != CRM_Core_Action::DELETE &&
-      (!$this->_id || !CRM_Core_DAO::getFieldValue('CRM_Core_BAO_MailSettings', $this->_id, 'activity_status'))
```

Comment:
This only condition in which `!CRM_Core_DAO::getFieldValue('CRM_Core_BAO_MailSettings', $this->_id, 'activity_status')` would apply is if you were switching an existing bounce processing mail account to an email-to-activity account, which would happen rarely-to-never, so I don't we need to handle this separately, since the consequence is low (the activity status would be scheduled by default instead of completed).

Reply:
* Another consequence besides what you mentioned about status is you end up with the default of only one of the from and recipients being recorded, and in general while not a technical problem, seems undesirable to allow. I.e. none of the people it was sent to get recorded.
* Also note that this situation then ends up being the default choices when creating an account for Oauth. So it's not as rare as thought.
* And personally changing from bounce to activity is something I do somewhat often since not all sites have bounce processing, but activity processing is used a lot in cases.

Otherwise though the PR overall looks pretty good. I haven't looked at the admin ui part. Seems there's a separate discussion there.

Reply:
I see what you mean, but not sure how to handle that. Could change the upgrade and the generated data to also set those defaults for bounce processing, which is fine because they don't do anything for bounce processing.
But that doesn't cover oauth. For that I think we'd need to add defaults in the oauth process or add defaults directly to the DB.
We could check if the activity status is not set and add defaults in that case, but that feels a bit odd and prone to create problems.
Thoughts?

I think the Admin UI part can go ahead as is. Hardly the only issue of that type in Admin UI.